recipes/lapack/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+import glob
+import os
+import shutil
+from io import StringIO
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanException
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class LapackConan(ConanFile):
+    name = "lapack"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/Reference-LAPACK/lapack"
+    description = "Fortran subroutines for solving problems in numerical linear algebra"
+    topics = (
+        "lapack"
+    )
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False], "visual_studio": [True, False]}
+    default_options = {"shared": False, "visual_studio": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    requires = "zlib/1.2.11"
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Macos" and \
+            self.settings.compiler == "apple-clang" and \
+            Version(self.settings.compiler.version.value) < "8.0":
+            raise ConanInvalidConfiguration("lapack requires apple-clang >=8.0")
+        if self.options.visual_studio and not self.options.shared:
+            raise ConanInvalidConfiguration("only shared builds are supported for Visual Studio")
+        if self.settings.compiler == "Visual Studio" and not self.options.visual_studio:
+            raise ConanInvalidConfiguration("This library needs option 'visual_studio=True' to be consumed")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
```

Comment:
Please cache the cmake object to avoid reconfiguration in `package` after `build`.
This can be doen by adding a `_cmake` attribute to the class.